.github/workflows/release.yml

```
@@ -1,43 +1,55 @@
-                          id = $1; \
-                          title = id.split("\/").last; \
-                          path = id.delete_prefix("/"); \
-                          url = "https://groonga.org/docs/#{path}.html"; \
-                          "[#{title}](#{url})"; \
-                        }. \
-                        gsub(/{ref}`(.+?)`/) {$1}.
-                        strip)' \
-             $(ls *.md | sort --human-numeric-sort | tail -n1)) > \
-            release-note.md
-          echo "" >> release-note.md
-          echo "### Translations" >> release-note.md
-          echo "" >> release-note.md
-          version=${GITHUB_REF_NAME#v}
-          major_version=${version%%.*}
-          version_hyphen=$(echo ${version} | tr . -)
-          echo "  * [Japanese](https://groonga.org/ja/docs/news/${major_version}.html#release-${version_hyphen})" >> release-note.md
-          title="$(head -n1 release-note.md | sed -e 's/^## //')"
-          tail -n +2 release-note.md > release-note-without-version.md
-          gh release create ${GITHUB_REF_NAME} \
-            --discussion-category Releases \
-            --notes-file release-note-without-version.md \
-            --title "${title}"
+          if gh release view "${{ inputs.tag }}" > /dev/null 2>&1; then
+            echo "Skip the creation step for Release ${{ inputs.tag }}."
+          else
+            (cd doc/source/news && \
+             ruby \
+               -e 'print("## Groonga "); \
+                   puts(ARGF.read.split(/^## Release /)[1]. \
+                          gsub(/^\(.+\)=$/, ""). \
+                          gsub(/{doc}`(.+?)`/) { \
+                            id = $1; \
+                            title = id.split("\/").last; \
+                            path = id.delete_prefix("/"); \
+                            url = "https://groonga.org/docs/#{path}.html"; \
+                            "[#{title}](#{url})"; \
+                          }. \
+                          gsub(/{ref}`(.+?)`/) {$1}.
+                          strip)' \
+               $(ls *.md | sort --human-numeric-sort | tail -n1)) > \
+              release-note.md
+            echo "" >> release-note.md
+            echo "### Translations" >> release-note.md
+            echo "" >> release-note.md
+            tag=${{ inputs.tag }}
+            version=${tag#v}
+            major_version=${version%%.*}inputs.tag
+            version_hyphen=$(echo ${version} | tr . -)
+            echo "  * [Japanese](https://groonga.org/ja/docs/news/${major_version}.html#release-${version_hyphen})" >> release-note.md
+            title="$(head -n1 release-note.md | sed -e 's/^## //')"
+            tail -n +2 release-note.md > release-note-without-version.md
+            gh release create ${{ inputs.tag }} \
+              --discussion-category Releases \
+              --notes-file release-note-without-version.md \
+              --title "${title}"
```

```suggestion
              --title "${title}" || : 
```

存在チェック無しで、このくらいで十分そう？雑？

`package` と `document` が同時に終わって両方で `gh release create` するかもしれないのは考えなくても良いか。発生しなさそう。

---

確かに、ほとんどコケることもないと思うのでシンプルな形で寄せておくのが良さそうですね。

---

fix: [1ae7b00](https://github.com/groonga/groonga/pull/2156/commits/1ae7b00395246bb497a27edafc7fbaf301464d07)